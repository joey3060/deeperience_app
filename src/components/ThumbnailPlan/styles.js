import { StyleSheet } from 'react-native'
import MainStyle from '../../styles'

module.exports = StyleSheet.create({
  tripView: {
    height: 200,
    paddingTop: 5,
    paddingBottom: 5,
    paddingLeft: 10,
    paddingRight: 10,
  },
  tripOverView: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingLeft: 15,
    paddingRight: 5,
    paddingTop: 10,
    paddingBottom: 10,
    borderBottomWidth: 1,
    borderColor: '#ccc',
  },
  backgroundImage: {
    flex: 1,
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
  },
  titleView: {
    width: 190,
    backgroundColor: 'rgba(0,0,0,0.5)',
    position: 'absolute',
    top: 10,
    left: 10,
    paddingLeft: 5,
  },
  title: {
    fontSize: MainStyle.font.big,
    color: 'white',
  },
  dayInfo: {
    color: 'white',
    backgroundColor: '#00A2FF',
    textAlign: 'center',
    fontSize: MainStyle.font.medium,
    position: 'absolute',
    top: 75,
    left: 10,
    paddingLeft: 5,
    paddingRight: 5,
  },
  tagView: {
    alignItems: 'center',
    position: 'absolute',
    bottom: 20,
    left: 10,
    flexDirection: 'row',
  },
  tags: {
    color: 'white',
    backgroundColor: '#01C33B',
    textAlign: 'center',
    fontSize: MainStyle.font.medium,
    paddingLeft: 5,
    paddingRight: 5,
    marginRight: 10,
  },
  statView: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  statIcon: {
    color: 'black',
    marginRight: 5,
  },
  statText: {
    color: 'black',
    fontSize: MainStyle.font.medium,
    marginRight: 10,
  },
  guideName: {
    fontSize: MainStyle.font.big,
    color: 'black',
  },
  price: {
    backgroundColor: 'rgba(0,0,0,0.5)',
    fontSize: MainStyle.font.big + 2,
    color: '#FF8000',
    fontWeight: 'bold',
    position: 'absolute',
    bottom: 20,
    right: 0,
    paddingLeft: 10,
    paddingRight: 10,
    textAlign: 'center',
  },
  avatar: {
    borderRadius: 35,
    borderColor: 'white',
    borderWidth: 2,
    height: 70,
    width: 70,
    position: 'absolute',
    top: 5,
    right: 5,
  },
})
