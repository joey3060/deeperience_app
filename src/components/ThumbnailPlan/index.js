'use strict'
import {
  Text,
  TouchableHighlight,
  Image,
  View,
} from 'react-native'
import styles from './styles'
import React, { Component, PropTypes } from 'react'
import Icon from 'react-native-vector-icons/FontAwesome'

class ThumbnailPlan extends Component {
  static propTypes = {
    title: PropTypes.string,
    dayInfo: PropTypes.string,
    guideName: PropTypes.string,
    starNum: PropTypes.number,
    watchNum: PropTypes.number,
    purchaseNum: PropTypes.number,
    price: PropTypes.number,
    unit: PropTypes.string,
    tags: PropTypes.array,
  }
  static defaultProps = {
    title: 'DeeperienceTrip',
    dayInfo: '1 Day',
    guideName: 'Deeperience',
    starNum: 0,
    watchNum: 0,
    purchaseNum: 0,
    price: 0,
    unit: 'TWD',
    tags: ['', ''],
  }

  render() {
    return (
      <TouchableHighlight>
        <View>
          <View style={styles.tripView}>
            <Image
              source={{ uri: this.props.backgroundImage }}
              style={styles.backgroundImage}
            />
            <View style={styles.titleView}>
              <Text
                numberOfLines={2}
                style={styles.titleOut}
              >
                <Text style={styles.title}>{this.props.title}</Text>
              </Text>
            </View>
            <Image
              source={{ uri: this.props.avatar }}
              style={styles.avatar}
            />
            <Text style={styles.dayInfo}>
              {this.props.dayInfo}
            </Text>
            <View style={styles.tagView}>
              <Text style={styles.tags}>
                {`${this.props.tags[0]}`}
              </Text>
              <Text style={styles.tags}>
                {`${this.props.tags[1]}`}
              </Text>
            </View>
            <Text style={styles.price}>
              {`${this.props.unit}${this.props.price}`}
            </Text>
          </View>
          <View style={styles.tripOverView}>
            <Text style={styles.guideName}>
              {this.props.guideName}
            </Text>
            <View style={styles.statView}>
              <Icon
                name="star"
                size={15}
                style={[styles.statIcon, { color: '#FFE600' }]}
              />
              <Text style={styles.statText}>
                {this.props.starNum}
              </Text>
              <Icon
                name="eye"
                size={15}
                style={[styles.statIcon, { color: '#FF00FF' }]}
              />
              <Text style={styles.statText}>
                {this.props.seenNum}
              </Text>
              <Icon
                name="shopping-cart"
                size={15}
                style={[styles.statIcon, { color: '#FF8000' }]}
              />
              <Text style={styles.statText}>
                {this.props.purchaseNum}
              </Text>
            </View>
          </View>
        </View>
      </TouchableHighlight>
    )
  }
}

export default ThumbnailPlan
