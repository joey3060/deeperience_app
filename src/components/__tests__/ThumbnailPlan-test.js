import React from 'react'
import ThumbnailPlan from '../ThumbnailPlan'
import renderer from 'react/lib/ReactTestRenderer'

describe('<ThumbnailPlan />', () => {
  it('should rendered correctly', () => {
    let props = {
      title: 'test',
      dayInfo: 'two days one night',
      guide: 'john',
      startNumber: '8',
      watchNumber: '7',
      numOfPurchase: '8787',
      cost: '7878',
      unit: 'TW',
    }
    const tree = renderer.create(<ThumbnailPlan {...props}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
})