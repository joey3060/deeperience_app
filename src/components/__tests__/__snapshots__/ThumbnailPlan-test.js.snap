exports[`<ThumbnailPlan /> should rendered correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function bound touchableHandleResponderGrant]}
  onResponderMove={[Function bound touchableHandleResponderMove]}
  onResponderRelease={[Function bound touchableHandleResponderRelease]}
  onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
  onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
  onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent"
      },
      undefined
    ]
  }
  testID={undefined}>
  <Image
    source={1}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderRadius": 10,
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "height": 130,
        "width": 355
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      lineBreakMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "height": 25,
          "marginLeft": 20,
          "marginTop": 15,
          "width": 175
        }
      }>
      test
    </Text>
    <Image
      source={1}
      style={
        Object {
          "borderColor": "white",
          "borderRadius": 25,
          "borderWidth": 1,
          "height": 50,
          "marginLeft": 90,
          "marginTop": 10,
          "width": 50
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      lineBreakMode="tail"
      style={
        Object {
          "borderColor": "white",
          "borderWidth": 1,
          "color": "white",
          "fontSize": 13,
          "height": 20,
          "marginLeft": 20,
          "marginTop": -15,
          "padding": 3,
          "width": 50
        }
      }>
      two days one night
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      lineBreakMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 13,
          "height": 20,
          "marginLeft": 10,
          "paddingLeft": 280,
          "width": 355
        }
      }>
      john
    </Text>
    <Text
      accessible={true}
      allowFontScaling={false}
      lineBreakMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "FontAwesome",
            "fontSize": 10,
            "fontStyle": "normal",
            "fontWeight": "normal"
          },
          Object {
            "color": "white",
            "marginLeft": 22,
            "marginTop": 30,
            "width": 10
          }
        ]
      }>
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      lineBreakMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 11,
          "marginLeft": 5,
          "marginTop": 28
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={false}
      lineBreakMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "FontAwesome",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "normal"
          },
          Object {
            "color": "white",
            "marginLeft": 10,
            "marginTop": 28,
            "width": 10
          }
        ]
      }>
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      lineBreakMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 11,
          "marginLeft": 5,
          "marginTop": 28
        }
      }>
      7
    </Text>
    <Text
      accessible={true}
      allowFontScaling={false}
      lineBreakMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "FontAwesome",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "normal"
          },
          Object {
            "color": "white",
            "marginLeft": 10,
            "marginTop": 28,
            "width": 10
          }
        ]
      }>
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      lineBreakMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 11,
          "marginLeft": 5,
          "marginTop": 28
        }
      }>
      8787
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      lineBreakMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 25,
          "marginLeft": 120,
          "marginTop": 13
        }
      }>
      7878
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      lineBreakMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 15,
          "marginTop": 21
        }
      }>
      TW
    </Text>
  </Image>
</View>
`;
